src/api/cpp/GravityNode.cpp

```diff
@@ -524,6 +539,105 @@ GravityNode::~GravityNode()
+	
+	// Create lists to hold sinks
+	std::list<shared_ptr<spdlog::sinks::sink>> app_sink_list = {};
+	std::list<shared_ptr<spdlog::sinks::sink>> gravity_sink_list = {};
+	
+	// Always create console loggers
+	auto shared_console_sink = make_shared<spdlog::sinks::stdout_color_sink_mt>();
+	auto console_proxy_for_gravity = std::make_shared<proxy_dist_sink_mt>();
+	console_proxy_for_gravity->add_sink(shared_console_sink);
+	console_proxy_for_gravity->set_level(gravity_console_level);
+	gravity_sink_list.push_back(console_proxy_for_gravity);
+	auto console_proxy_for_app = std::make_shared<proxy_dist_sink_mt>();
+	console_proxy_for_app->add_sink(shared_console_sink);
+	console_proxy_for_app->set_level(app_console_level);
+	app_sink_list.push_back(console_proxy_for_app);
+	
+	// Configure file logger (if specified)
+	if (has_gravity_file_logger || has_app_file_logger)
+	{
+		string filename = getStringParam("LogDirectory", ".") + file_separator + componentID + ".log";
+		auto sharedFileSink = std::make_shared<spdlog::sinks::basic_file_sink_mt>(filename);
+		
+		if (has_gravity_file_logger)
+		{
+			auto file_proxy_for_gravity = std::make_shared<proxy_dist_sink_mt>();
+			file_proxy_for_gravity->add_sink(sharedFileSink);
+			file_proxy_for_gravity->set_level(gravity_file_level);
+			gravity_sink_list.push_back(file_proxy_for_gravity);
+		}
+		
+		if (has_app_file_logger)
+		{
+			auto file_proxy_for_app = std::make_shared<proxy_dist_sink_mt>();
+			file_proxy_for_app->add_sink(sharedFileSink);
+			file_proxy_for_app->set_level(app_file_level);
+			app_sink_list.push_back(file_proxy_for_app);
+		}
+		
+	}
+	
+	// Configure publish logger
+	if (has_app_publish_logger)
+	{
+		auto publishSink = std::make_shared<PublishSink<std::mutex>>(this);
+		auto publish_proxy_for_app = std::make_shared<proxy_dist_sink_mt>();
+		publish_proxy_for_app->add_sink(publishSink);
+		publish_proxy_for_app->set_level(app_publish_level);
+		app_sink_list.push_back(publish_proxy_for_app);
+	}
+	
+	// Configure GravityLogger.
+	logger = std::make_shared<spdlog::logger>("GravityLogger");
+	for (auto sink : gravity_sink_list)
+	{
+		logger->sinks().push_back(sink);
+	}
+	logger->set_level(spdlog::level::trace); // logger will pass through all logs to be filtered by sinks
+	logger->flush_on(spdlog::level::trace); // logger will flush on all messages
+	logger->set_pattern("[%m/%d/%Y %T.%f " + componentID + "-%l] %v");
+	spdlog::register_logger(logger);
+	
+	// Configure ApplicationLogger
+	auto app_logger = std::make_shared<spdlog::logger>("GravityApplicationLogger");
+	for (auto sink : app_sink_list)
+	{
+		app_logger->sinks().push_back(sink);
+	}
+	app_logger->set_level(spdlog::level::trace); // logger will pass through all logs to be filtered by sinks
+	app_logger->flush_on(spdlog::level::trace); // logger will flush on all messages
+	app_logger->set_pattern("[%m/%d/%Y %T.%f " + componentID + "-%l] %v");
```

Comment:
Is this to match existing format? I otherwise would push for Y-M-d - handy if/when sorting or merging logs.  Not a major use case though.

Reply:
The intent was to match the existing format but I'm open to changing it to something more useful if there are no objections. It is also possible now for the applications to set their own desired logging format.

Reply:
Up to you guys, I use Gravity less than you and @mbarger99 by a good margin.  I always lean towards Ymd whenever I can.

Reply:
Agreed. I use YMD everywhere else too. I'll make the mod.